src/main/java/seedu/momentum/ui/ProjectCard.java

```diff
@@ -35,32 +33,37 @@
     @FXML
     private Label id;
     @FXML
-    private Label description;
+    private HBox description;
     @FXML
-    private Label createdDate;
+    private HBox createdDate;
     @FXML
-    private Label deadline;
+    private HBox deadline;
     @FXML
     private FlowPane tags;
-    @FXML
-    private StackPane durationListPanelPlaceholder;
 
     /**
-     * Creates a {@code ProjectCode} with the given {@code Project} and index to display.
+     * Creates a {@code ProjectCard} with the given {@code Project} and index to display.
      */
     public ProjectCard(Project project, int displayedIndex) {
         super(FXML);
         this.project = project;
         id.setText(displayedIndex + ". ");
         name.setText(project.getName().fullName);
-        description.setText(project.getDescription().value);
-        createdDate.setText(project.getCreatedDate().getFormatted());
-        deadline.setText(project.getDeadline().getFormattedDeadline());
+
+        if (project.hasDescription()) {
+            description.getChildren().add(new Label(project.getDescription().value));
+        }
+
+        createdDate.getChildren().add(new Label(project.getCreatedDate().getFormatted()));
+
+        String deadlineText = project.hasDeadline()
+            ? project.getDeadline().getFormattedDeadline()
+            : "No deadline set";
```

Comment:
Maybe change the getFormatted method in deadline to return no deadline set when there is no deadline instead so that a project does not have to know whether it has a deadline.

Reply:
I think this is a good suggestion, you can remove the hasDeadline() method by doing this!